DigitalOffice/Services/TimeManagementService/Validators/CreateWorkTimeRequestValidator.cs

```diff
@@ -0,0 +1,93 @@
+﻿using FluentValidation;
+using Microsoft.AspNetCore.Mvc;
+using System;
+using LT.DigitalOffice.TimeManagementService.Models;
+using LT.DigitalOffice.TimeManagementService.Repositories.Filters;
+using LT.DigitalOffice.TimeManagementService.Repositories.Interfaces;
+
+namespace LT.DigitalOffice.TimeManagementService.Validators
+{
+    public class CreateWorkTimeRequestValidator : AbstractValidator<CreateWorkTimeRequest>
+    {
+        private IWorkTimeRepository repository;
+
+        /// <summary>
+        /// How many days ago can WorkTime be added.
+        /// </summary>
+        public readonly static int fromDay = 3;
+        /// <summary>
+        /// How many days ahead can WorkTime be added.
+        /// </summary>
+        public readonly static int toDay = 2;
+        /// <summary>
+        /// Limit on working hours in a row.
+        /// </summary>
+        public readonly static TimeSpan workingLimit = new TimeSpan(24, 0, 0); // 24 hours.
+
+        private readonly DateTime fromDateTime = DateTime.Now.AddDays(-fromDay);
+        private readonly DateTime toDateTime = DateTime.Now.AddDays(toDay);
+
+        public CreateWorkTimeRequestValidator([FromServices] IWorkTimeRepository repository)
+        {
+            this.repository = repository;
+
+            RuleFor(wt => wt.WorkerUserId)
+                    .NotEmpty();
+
+            RuleFor(wt => wt.Date)
+                .NotEmpty()
+                .When(wt => wt.Date > fromDateTime)
+                .WithMessage(wt => string.Format(
+                    "WorkTime had to be filled no later than {0}.",
+                    fromDateTime.ToString(System.Globalization.CultureInfo.CreateSpecificCulture("en-GB"))))
+                .When(wt => wt.Date < toDateTime)
+                .WithMessage(wt => string.Format(
+                    "WorkTime cannot be filled until {0}.",
+                    fromDateTime.ToString(System.Globalization.CultureInfo.CreateSpecificCulture("en-GB"))));
+
+            RuleFor(wt => wt.Time)
+                .NotEmpty()
+                .When(wt => wt.Time.TotalMinutes <= 0)
```

Comment:
I think, here TotalMinutes should be > 0.

Reply:
Fixed.